File: Easy/lowest-common-ancestor-of-a-binary-search-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */

class Solution {        // 6ms
    public TreeNode lowestCommonAncestor(TreeNode root, TreeNode p, TreeNode q) {
        int small = Math.min(p.val, q.val);
        int big = Math.max(p.val, q.val);
        TreeNode ret = root;
        
        while(small > root.val || big < root.val){
            if(small > root.val){
                root = root.right;
            } else {
                root = root.left;
            }
            ret = root;
        }
        return ret;
    }
}

File: Easy/lowest-common-ancestor-of-a-binary-search-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:     # 76ms
    def lowestCommonAncestor(self, root: 'TreeNode', p: 'TreeNode', q: 'TreeNode') -> 'TreeNode':
        small = min(p.val, q.val)
        big = max(p.val, q.val)
        while small > root.val or big < root.val:
            root = root.right if small > root.val else root.left
        return root